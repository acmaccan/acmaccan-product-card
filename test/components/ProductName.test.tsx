import React from 'react';
import renderer from 'react-test-renderer';
import { ProductName, ProductCard } from '../../src/components';
import { product1 } from '../data/products';

describe('Product Name', () => {
  test('Should render the component with its custom name', () => {
    const wrapper = renderer.create(<ProductName name="Custom Name" />);
    // console.log(wrapper.toJSON());
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Should render the component with its default name', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>{() => <ProductName />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
