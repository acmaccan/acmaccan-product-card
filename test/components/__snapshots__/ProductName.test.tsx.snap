// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Name Should render the component with its custom name 1`] = `
<span
  className="productDescription undefined"
>
  Custom Name
</span>
`;

exports[`Product Name Should render the component with its default name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Without Imagen
  </span>
</div>
`;
